File: problems/2-medium/128-Longest-Consecutive-Sequence/solution.cpp
#include <vector>
#include <unordered_set>

using namespace std;


int longestConsecutive(vector<int>& nums) {
    int max_length = 0;
    unordered_set<int> nums_set(nums.begin(), nums.end());
    for (int num: nums_set) {
        if (nums_set.find(num-1) == nums_set.end()) {
            int length = 0;
            while (nums_set.find(num) != nums_set.end()) {
                num++;
                length++;
            }
            max_length = max(max_length, length);
        }
    }
    return max_length;
}

File: problems/2-medium/128-Longest-Consecutive-Sequence/solution.java
import java.util.HashSet;
import java.util.Set;

public int longestConsecutive(int[] nums) {
    int maxLength = 0;
    Set<Integer> numsSet = new HashSet<>();
    
    // Add all numbers to set
    for (int num : nums) {
        numsSet.add(num);
    }
    
    for (int num : numsSet) {
        // Only start counting if this is the beginning of a sequence
        if (!numsSet.contains(num - 1)) {
            int length = 0;
            while (numsSet.contains(num)) {
                num++;
                length++;
            }
            maxLength = Math.max(maxLength, length);
        }
    }
    
    return maxLength;
}